paraglob.cpp

```diff
@@ -0,0 +1,111 @@
+#include "paraglob.h"
+
+Paraglob::Paraglob(const std::vector<std::string> &patterns): Paraglob() {
+  for (std::string pattern : patterns) {
+    this->add(pattern);
```

Comment:
`this->` seems like an unusual style, any particular reason for using that?

Reply:
This is just a stylistic thing that I've picked up over time. I tend to think that for functions with generic names like add its better to be explicit that I'm calling the classes method. In my personal opinion as well I think code like this (while a little more verbose) is a little more clear for reading. If you think its excessively verbose though I've got no problem getting rid of it :)